File: solutions/t9-spelling/dart/approach3.dart
import 'dart:io';

main() {

  var alphabet = [" ", "", "abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"];

  new File('A-small-practice.in').readAsLines().then((List<String> lines) {
    
    var n = int.parse(lines[0]);

    for (var i = 1; i < n; i++){

        var anterior = -1;
        var palabra = lines[i];
        stdout.write("Case #${i}: ");

        for(var j = 0; j < palabra.length; j++) {
            for(var k = 0; k < alphabet.length; k++) {
                for(var l = 0; l < alphabet[k].length; l++) {

                if(palabra[j].toString() == alphabet[k][l]) {

                    if (anterior == k) {

                        stdout.write(" ");

                    }

                    for (var m = 0; m <= l; m++) {

                    stdout.write(k);
                    anterior = k;

                    }

                }

                }
            }
        }
    print("");

    }
  });
    }

File: solutions/t9-spelling/scala/approach3.scala
object Problema3 {
   def main(args: Array[String]) {
     
        val source = scala.io.Source.fromFile("A-large-practice.in")
        val lines = source.getLines
        val n = lines.next.toInt

        var alphabet = Array(" ", "", "abc","def","ghi","jkl","mno","pqrs","tuv","wxyz")

        for (i <- 0 until n) {
            var anterior = -1
            var palabra = lines.next

            print("Case #"+(i+1)+": ")

        for (j <- 0 until palabra.size) {
        
            for (k <- 0 until alphabet.length) {
                
                for (l <- 0 until alphabet(k).size){

                    if(palabra.charAt(j).equals(alphabet(k).charAt(l))){

                        if (anterior == k){
                            print(" ")
                        }

                        for(m <- 0 to l) {
                            print(k)
                        }

                        anterior = k

                    }

                }

            }
        }
            println("")
        }
    }
}
